### PR TITLE
Incorporate dump1090-mutability; improved signal strength reporting; bug fixes

### PR DESCRIPTION
**Pull request by AvSquirrel**
Improvements:

- Incorporate dump1090 changes from #285 into a recent branch of dump1090-mutability. This is an active branch of dump1090, and has added additional functionality over the MalcolmRobb branch,
- Traffic UI now shows if ground-based traffic transmissions are TIS-B or ADS-R on both UAT and 1090ES.
- Improvements to signal strength reporting. All information sources (1090ES traffic, UAT traffic, and UAT FIS-B uplinks) are now displayed with received power converted to dB and scaled the same. Minimum signal detection occurs at about -40 dB, with various messages detectable at higher powers. Short Mode S messages (e.g. interrogation replies) are the easiest to detect, and can be seen starting at about -35 dB, and ADS-B / FIS-B type messages will be seen starting at -30 dB, and reliably above -25 dB. Nominal saturation occurs at 0.0 dB, and the strongest signals will truncate at about +1.4 dB.
- Create and check for dump1090 heartbeat message
- Towers page refreshes more quickly, and shows current, average, and peak power (dB) levels. (Old traffic page shows strength as sum of amplitude.)
- Allow UAT emitters with NIC = 0 into the traffic data stream. This will make uncertified and/or misconfigured UAT transmitters visible.
- Assorted bug fixes

To-do:
- Validate performance of new 1090ES TIS-B support
- Add CSV-out option (or converter tool) for translating JSON-formatted traffic into a format easier for @Axtel4 to parse
- "Signal bar" mappings for dB signal strength. 
- Merge dump1090 improvements into mutability's master branch

image

image

### CODE REVIEW

File: main/traffic.go [hunk 1]
```diff
@@ -835,6 +845,17 @@ func updateDemoTraffic(icao uint32, tail string, relAlt float32, gs float64, off
 	if ti.Addr_type == 1 {         //reserved value
 		ti.Addr_type = 0
 	}
+
+	if ti.Addr_type == 0 {
+		ti.TargetType = TARGET_TYPE_ADSB
+	}
+	if ti.Addr_type == 2 {
```
Comment (jpoirier) [inline main/traffic.go]: these two checks can be done with an "if else" check 

I guess if ti.Addr_type isn't 0 or 2 or 3 then the default for ti.TargetType is TARGET_TYPE_MODE_S (TARGET_TYPE_MODE_S is 0 which is what ti.TargetType is initialized to). It'd be clearer if the last check was an else that explicitly defaults to the desired value. E.g.

        if ti.Addr_type == 0 {
		ti.TargetType = TARGET_TYPE_ADSB
	} else if ti.Addr_type == 2 {
		ti.TargetType = TARGET_TYPE_ADSR
	} else if ti.Addr_type == 3 {
		ti.TargetType = TARGET_TYPE_TISB
	} else {
                ti.TargetType = TARGET_TYPE_MODE_S // or should this be something else
        }

**Comment by AvSquirrel** [comment]
Flew today, and realized I'm not handling the CA field in dump1090-mutability quite right for DF18 traffic. Stand by for a new dump1090 submodule update.

@jpoirier -- noted. Though it is a function I'm using solely to generate "demo" targets (and therfore the default case isn't too critical) it's worth cleaning up.

File: main/traffic.go (hunk 1 above)
Comment (AvSquirrel) [inline main/traffic.go]: Addressed with 769169d; copied logic from `parseDownlinkReport()`

**Comment by AvSquirrel** [comment]
CA/CF fix included on commit 769169d. Validated against the airborne message types (DF0, DF4, DF5, DF11, DF17); may need to flight test with a 1090-In airplane to validate DF18.

**Comment by cyoung** [comment]
@AvSquirrel - Q-bit handling in this branch? Have not reviewed the mutability branch at all, let's give it a try anyway. Did you enable oversampling?

https://github.com/cyoung/stratux/issues/171#issuecomment-167903450

**Comment by AvSquirrel** [comment]
Ayup.

Oversampling is enabled in the exec call in sdr.go, and q-bit is supported. (The mutability branch is based on the last the most MalcolmRobb commit, and is now 497 commits ahead of it.)

**Comment by cyoung** [comment]
Aha. Sounds like a winner.

**Pull request merged**